spharpy/spherical.py

```diff
@@ -1,6 +1,431 @@
+
+        The inverse transform matrix is calculated based on the specified `inv_transform_type`.
+        If 'pseudo_inverse' is chosen, the Moore-Penrose pseudo-inverse is used.
+        If 'quadrature' is chosen, the inverse is computed using the conjugate transpose of
+        the basis matrix multiplied by 4 * π * weights.
+
+        Parameters
+        ----------
+        inv_transform_type : {'pseudo_inverse', 'quadrature'}, optional
+            Type of inverse transform to compute.
+        weights : array-like, optional
+            Sampling weights for the quadrature transform. Required if 'quadrature' is chosen.
+
+        Returns
+        -------
+        None
+        """
+        if self.basis is None:
+            self.compute_basis()
+        if inv_transform_type != self.inv_transform_type:
+            assert inv_transform_type in ['pseudo_inverse', 'quadrature'], (
+                "Invalid inverse transform type, "
+                "currently only 'pseudo_inverse' "
+                "and 'quadrature' are supported")
+            self.inv_transform_type = inv_transform_type
+        # print("computing inverse basis using ", self.inv_transform_type)
+        if self.inv_transform_type is None:
+            ValueError("Inverse transform type not specified")
+        elif self.inv_transform_type == 'pseudo_inverse':
+            self._basis_inv = np.linalg.pinv(self.basis)
+        elif self.inv_transform_type == 'quadrature':
+            if weights is None:
+                print("Warning: No weights specified for quadrature transform,"
+                      "calculating weights using voronoi tessellation of sphere")
+                weights = calculate_sampling_weights(self.coords)
+            self._basis_inv = np.conj(self.basis).T * (weights)
+
+    @property
+    def basis_inv_gradient_theta(self):
+        if self._basis_inv_gradient_theta is None or self._basis_inv_gradient_phi is None:
+            self.compute_inverse_gradient()
+        return self._basis_inv_gradient_theta
+
+    @property
+    def basis_inv_gradient_phi(self):
+        if self._basis_inv_gradient_theta is None or self._basis_inv_gradient_phi is None:
+            self.compute_inverse_gradient()
+        return self._basis_inv_gradient_phi
+
+    def compute_inverse_gradient(self, inv_transform_type=None, weights=None):
+        if inv_transform_type != self.inv_transform_type:
+            self.inv_transform_type = inv_transform_type
+        if self.inv_transform_type is None:
+            ValueError("Inverse transform type not specified")
+        elif self.inv_transform_type == 'pseudo_inverse':
+            self._basis_inv_gradient_theta = np.linalg.pinv(self.basis_gradient_theta)
+            self._basis_inv_gradient_phi = np.linalg.pinv(self.basis_gradient_phi)
+        elif self.inv_transform_type == 'quadrature':
+            if weights is None:
+                weights = calculate_sampling_weights(self.coords)
```

Comment:
Maybe during setting/init the ``spharpy.SamplingSphere``, we could check if it contains ``weigths`` and ``n_max``, otherwise raise an error. Then the user need to take care of it.